### PR TITLE
Add Support for Password and Fingerprint Authentication

### PR DESCRIPTION
**Pull request by milouk**
This commit adds password protection to the App. Moreover, if available the user can also login via fingerprint.

### CODE REVIEW

**Comment by milouk** [comment]
- `SwirlView `provides the animated fingerprint icon which you can see once you enable fingerprint authentication. For more info check here https://github.com/mattprecious/swirl

-  Yeah about the newlines you are correct, ill fix it asap. Ditto for the codestyling.

**Comment by MSF-Jarvis** [comment]
Have you actually tested it? Settings crashed on me immediately... Here's the diff to resolve it https://del.dog/raw/ojesoyucut

**Comment by milouk** [comment]
Of course i have tested it. Thanks for the diff. I already uploaded the patch.

**Comment by MSF-Jarvis** [comment]
I still haven't looked too closely at the code but the UI/UX flow here needs a lot of work. Enforcing the presence of a text password to use the fingerprint feels rather awful to me. There is a very functional BiometricPrompt both in the [framework](https://developer.android.com/reference/android/hardware/biometrics/BiometricPrompt) as well as in [AndroidX](https://developer.android.com/reference/androidx/biometric/BiometricPrompt) which I would much prefer over manually putting together this cludge we have right now.

So, to summarize, here's my suggestions for the UX of this pull request:
- Don't cobble together the fingerprint prompt, instead use framework/AndroidX provided APIs.
- Don't enforce the presence of a text password for enabling fingerprint authentication.
- Use Material Components elements to build the UI elements you've added, they're nicer than plain TextViews and Buttons.

**Comment by milouk** [comment]
The reason fingerprint authentication requires another form of authentication is in case there are too many invalid attempts which result in the user getting locked out of the app. The same behavior exists in almost everything that includes a fingerprint authentication e.g Android System Unlocking where you can't setup a fingerprint unlock security unless you have first configured another form of authentication e.g password, PIN etc.

Moreover, ill try to convert it to BiometricPrompt and improve the UI to comply with Material Design guidelines.

**Comment by msfjarvis** [comment]
This has been dormant for far too long, closing in favour of #541.
